lib/syntax-definition/src/Language/Symbol/Operator/Prec/Poset.hs

```diff
@@ -0,0 +1,175 @@
+module Language.Symbol.Operator.Prec.Poset where
+
+import Prologue
+
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+
+import Control.Lens (non, (?~))
+import Data.Map     (Map)
+import Data.Set     (Set)
+
+-----------------------
+-- === Partition === --
+-----------------------
+
+-- === Definition === --
+
+data Partition n = Partition
+    { _representantsMap :: Map n n
+    , _membersMap       :: Map n (Set n)
+    } deriving (Show, Eq)
+makeLenses ''Partition
+
+instance Default (Partition n) where
+    def = Partition def def
+
+-- === Private API === --
+
+setRepresentant :: Ord n => n -> n -> Partition n -> Partition n
+setRepresentant rep item partition = let
+    fixedRepMap     = partition & representantsMap . at item ?~ rep
```

Comment:
Indentation here is a bit strange. No point lining things up if you wrap the next line!